File: cmd/faas/templates/hello-node/handler.js
// Hello-world handler for faas.
//
// Listens on :8080 (the port guest-init forwards to). Returns a tiny
// JSON greeting so a curl check is enough to verify a deploy landed.
// The handler also reads a few env vars (set via `faas env push`) to
// show how secrets surface inside the guest — values are NEVER
// returned in the response, only the key names, so logging them
// doesn't leak plaintext.

import express from "express";

const app = express();
const port = process.env.PORT || 8080;

app.get("/", (_req, res) => {
  res.json({
    message: "hello from faas",
    node: process.version,
    secretKeys: Object.keys(process.env)
      .filter((k) => !k.startsWith("FAAS_") && k !== "PATH" && k !== "HOME" && k !== "NODE_VERSION")
      .sort(),
  });
});

app.get("/healthz", (_req, res) => {
  res.status(200).json({ ok: true });
});

app.listen(port, () => {
  console.log(`hello-faas listening on :${port}`);
});

File: cmd/faas/templates/hello-python/handler.py
"""Hello-world handler for faas (Flask).

Listens on :8080 (the port guest-init forwards to). Returns a tiny
JSON greeting so a curl check is enough to verify a deploy landed.
The handler reads env vars (set via `faas env push`) and reports
the key NAMES only — never values — so a stdout log scrape can't leak
plaintext.
"""
import os

from flask import Flask, jsonify

app = Flask(__name__)


@app.get("/")
def root():
    # Skip env vars the runner sets itself; surface only customer keys.
    skip = {"PATH", "HOME", "LANG", "PYTHONUNBUFFERED", "FAAS_APP", "FAAS_DEPLOY"}
    secret_keys = sorted(
        k for k in os.environ.keys()
        if not k.startswith("FAAS_") and k not in skip
    )
    return jsonify(
        message="hello from faas",
        python=os.environ.get("PYTHON_VERSION", "unknown"),
        secret_keys=secret_keys,
    )


@app.get("/healthz")
def healthz():
    return jsonify(ok=True)


if __name__ == "__main__":
    port = int(os.environ.get("PORT", "8080"))
    app.run(host="0.0.0.0", port=port)  # noqa: S104 — guest-only listener